code/glTF2Asset.h

```diff
@@ -232,7 +230,7 @@ namespace glTF2
             case ComponentType_UNSIGNED_BYTE:
                 return 1;
             default:
-                throw DeadlyImportError("GLTF: Unsupported Component Type "+t);
+                throw DeadlyImportError("GLTF: Unsupported Component Type " + std::to_string(t));
```

Comment:
Clang was warning about  simply appending `t` to the string with the + sign. Is this better?

Reply:
Yes. The literal `"GLTF: Unsupported Component Type "` is a C-style string, which is a character pointer (`char *`) with the convention that the end of the string is indicated by a null byte. These are sometimes called "null-terminated strings".

So, suppose the incoming `ComponentType` `t` happens to be `ComponentType_BYTE`, which we've specified to have an underlying integer value of `5120`. Then the expression `"my C-style string" + t` consists in adding an `enum ComponentType` to a `char *`, and the result will be of the form `<address of my c-style string> + 5120`, which will cause the `DeadlyImportError` to be initialized with whatever happens to be in memory at that location, reading until it finds a null byte.

If, however, we explicitly convert `t` to a `std::string`--an object provided by the C++ standard library in the `<string>` header--as in the expression `"my c-style string" + std::to_string(t)`, the compiler uses the `+` operator overloaded by the `std::string` class, and the result is a `std::string` of the form `"my c-style string<whatever's in t>"`.